core/src/main/java/org/elasticsearch/search/aggregations/bucket/sampler/InternalSampler.java

```diff
@@ -48,11 +48,6 @@ public String getWriteableName() {
     }
 
     @Override
-    protected String getType() {
-        return "sampler";
```

Comment:
This is weird and prevented the tests from working. We implement this method in `InternalAggregation#getType()` to return `getWriteableName()`, and InternalSampler and its subclass UnmappedSampler have different names there. If we used "sampler" for both classes in the XContent output as type prefix we wouldn't be able to distinguish the two when parsing. I wonder if this was added by mistake or for a reason, maybe you can double check this?